src/cpp/modules/finite/finite.h

```diff
@@ -117,7 +117,7 @@ class FiniteDifferencesEngine
             tmp_b = originalInput - delta;
             T dx = delta * 2;
             tmp_f = tmp_b + dx;
-            // adjusting dx so that (tmp_b + dx) - tmp_b == delta
+            // adjusting dx so that (tmp_b + dx) - tmp_b == 2 * delta
```

Comment:
```
// adjusting dx so that (tmp_b + dx) - tmp_b == dx
```
In order to be absolutely clear it's about associativity.   As now, the reader needs to find the definition of dx, and to reassure themselves that dx == delta*2 exactly.

Reply:
Agreed.